pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -275,6 +275,20 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 	// Now that everything is synced, we can perform the actual processing
 	result, err := c.handleScaling(ctx, podAutoscaler, &podAutoscalerInternal)
 
+	// Update current replicas
+	targetGVK, targetErr := podAutoscalerInternal.TargetGVK()
+	if targetErr != nil {
+		podAutoscalerInternal.SetError(targetErr)
+		return autoscaling.NoRequeue, targetErr
```

Comment:
In that case you'd miss status update

Reply:
Added a check to update status and return the error at the end; priority for error during scaling > error to get target > error when updating status

Reply:
Makes sense, I can wait for the changes from https://github.com/DataDog/datadog-agent/blob/f35a065b1e4c9f735ea00edceba0718f249bbe5a/pkg/clusteragent/autoscaling/workload/controller.go#L413-L427
to be merged so I can make a status update and return early once `targetErr` is encountered